bgpd/bgp_evpn.c

```diff
@@ -504,8 +505,10 @@ static void map_vni_to_rt(struct bgp *bgp, struct bgpevpn *vpn,
 			/* Already mapped. */
 			return;
 
-	if (!irt)
+	if (!irt) {
 		irt = import_rt_new(bgp, &eval_tmp);
+		bgp_rtc_add_static(bgp, eval, BGP_RTC_MAX_PREFIXLEN);
```

Comment:
I always add a static RTC route regardless if the address family is configured for any peer. Is this problematic and if it is, is there a way to check if an address family is configured for at least one peer? I could also add a bgp option like `advertis-all-vni` that toggles the behavior of automatically generating RTC routes. Maybe even independent for l2vpn and l3vpn.

Reply:
Hi @sohn123, please separate both variables: advertisement versus populating.
- you populate a static RTC route in the BGP RIB.
- you advertise to bgp peers the RTC route if the matching afi/safi is enabled

I think RTC will be enabled if either L2VPN or L3VPN is enabled on neighbors.
So RTC will be sent in this case.
I know you implemented something for L2VPN, and not for L3VPN.
I would say RTC should be common for both cases:  are there many cases where both L3VPN and L2VPN are used?

about https://github.com/FRRouting/frr/pull/13476#issuecomment-1672967028,
I understand, RTC routes will have to be modified when dealing with iBGP, so I think this is something that will be done in bgp_updgrp_packet.c most probably. There is something done like that for BGP LU with label replacement.
I think this is something that can be done separately.